src/libguac/thread-local.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "config.h"
+#include "guacamole/thread-local.h"
+
+#include <stdlib.h>
+#include <pthread.h>
+#include <errno.h>
+#include <sched.h>
+
+/**
+ * Maximum number of thread-local keys supported.
+ */
+#define MAX_THREAD_KEYS 1024
```

Comment:
If the issue is partly:

> scalability limitations where systems are typically limited to 1024 pthread keys per process

doesn't this default leave the same limitation in place by default?